agreementbot/secure_api.go

```diff
@@ -543,7 +544,17 @@ func (a *SecureAPI) writeCompCheckResponse(w http.ResponseWriter, output interfa
 // Handles secret fetch, updates and delete using the vault API
 func (a *SecureAPI) secrets(w http.ResponseWriter, r *http.Request) {
 
-	ec, msgPrinter, userAuthenticated := a.processUserCred("/org/{org}/secrets/{secret}", w, r)
+	// Process in the inputs and verify that they are consistent with the logged in user.
+	pathVars := mux.Vars(r)
+	org := pathVars["org"]
+	user := pathVars["user"]
+	vaultSecretName := pathVars["secret"]
+	api_url := "/org/%v/secrets/%v"
+	if user != "" {
+		api_url = "/org/%v/secrets/user/%v/%v"
+	}
+
+	ec, msgPrinter, userAuthenticated := a.processUserCred(fmt.Sprintf(api_url, "{org}", "{user}", "{secret}"), w, r)
```

Comment:
How does this work for the defualt api_url case? the string {user} will be inserted after /secrets/ in the default URL string. Same problem on line 575.